evm/utils/test_builder/builder_utils.py

```diff
@@ -0,0 +1,74 @@
+import copy
+import random
+import subprocess
+
+from collections.abc import (
+    Mapping,
+)
+
+from evm.db.backends.memory import MemoryDB
+
+from eth_utils import (
+    force_text,
+    int_to_big_endian,
+)
+
+from eth_keys import keys
+
+
+def merge_nested(*dicts):
```

Comment:
*functional nonsense*

Slightly more concise and uses fully non-mutation based approach

```python
from cytoolz import merge_with

def merge_if_dicts(values):
    if all(isinstance(item, Mapping) for item in values):
        return merge_with(merge_if_dicts, *values)
    else:
        return values[-1]

def deep_merge(*dicts):
    return merge_with(merge_if_dicts, *dicts)
```

Reply:
👍 for `deep_merge` over `merge_nested`, as its analogous to `copy.deepcopy`